php-frontend/src/main/java/org/sonar/plugins/php/api/symbols/SymbolTable.java

```diff
@@ -17,32 +17,40 @@
  * License along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
  */
-package org.sonar.php.checks.utils;
+package org.sonar.plugins.php.api.symbols;
 
-import com.sonar.sslr.api.AstNode;
+import com.google.common.annotations.Beta;
+import com.google.common.collect.ImmutableList;
+import org.sonar.php.tree.symbols.Scope;
+import org.sonar.plugins.php.api.tree.Tree;
 
-public class Variable {
-  private final AstNode declaration;
-  private int usage = 0;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Set;
 
-  public Variable(AstNode declaration) {
-    this.declaration = declaration;
-  }
+@Beta
+public interface SymbolTable {
+  /**
+   * Returns all symbols in script
+   */
+  ImmutableList<Symbol> getSymbols();
 
-  public Variable(AstNode declaration, int usage) {
-    this.declaration = declaration;
-    this.usage = usage;
-  }
+  /**
+   *
+   * @param kind kind of symbols to look for
+   * @return list of symbols with the given kind
+   */
+  List<Symbol> getSymbols(Symbol.Kind kind);
 
-  public void increaseUsage() {
-    this.usage++;
-  }
+  /**
+   *
+   * @param name name of symbols to look for
+   * @return list of symbols with the given name
+   */
+  List<Symbol> getSymbols(String name);
```

Comment:
@vilchik-elena This method is only used in tests: do we need to have it in the public API? (The same question is also valid for `getSymbols()`.)